spec/util/subscribeToResult-spec.ts

```diff
@@ -0,0 +1,156 @@
+import {expect} from 'chai';
+import * as Rx from '../../dist/cjs/Rx';
+import {subscribeToResult} from '../../dist/cjs/util/subscribeToResult';
+import {OuterSubscriber} from '../../dist/cjs/OuterSubscriber';
+import {$$iterator} from '../../dist/cjs/symbol/iterator';
+import $$symbolObservable from 'symbol-observable';
+
+describe('subscribeToResult', () => {
+  it('should immeidately complete when subscribe to scalarObservable', (done: MochaDone) => {
+    const result = Rx.Observable.of(42);
+    const subscriber = new OuterSubscriber(x => {
+      expect(x).to.be.equal(42);
+    }, () => {
+      done(new Error('should not be called'));
+    }, done);
+
+    const subscription = subscribeToResult(subscriber, result);
+    expect(subscription).to.be.null;
+  });
+
+  it('should subscribe to observable', (done: MochaDone) => {
+    const expected = [1, 2, 3];
+    const result = Rx.Observable.range(1, 3);
+
+    const subscriber = new OuterSubscriber(x => {
+      expect(expected.shift()).to.be.equal(x);
+    }, () => {
+      done(new Error('should not be called'));
+    }, () => {
+      expect(expected).to.be.empty;
+      done();
+    });
+
+    subscribeToResult(subscriber, result);
+  });
+
+  it('should emits error when observable emits error', (done: MochaDone) => {
+    const result = Rx.Observable.throw(new Error('error'));
+    const subscriber = new OuterSubscriber(x => {
+      done(new Error('should not be called'));
+    }, (err) => {
+      expect(err).to.be.an('error', 'error');
+      done();
+    }, () => {
+      done(new Error('should not be called'));
+    });
+
+    subscribeToResult(subscriber, result);
+  });
+
+  it('should subscribe to array', (done: MochaDone) => {
+    const result = [1, 2, 3];
+    const expected = [];
+
+    const subscriber = new OuterSubscriber(x => expected.push(x),
+    () => {
+      done(new Error('should not be called'));
+    }, () => {
+      expect(expected).to.be.deep.equal(result);
+      done();
+    });
+
+    subscribeToResult(subscriber, result);
```

Comment:
Same issue as above, you'll want to make the test synchronous and assert right after the call to `subscribeToResult`.